packages/cashc/test/valid-contract-files/date_literal.cash

```diff
@@ -0,0 +1,8 @@
+pragma cashscript ^0.5.0;
+
+contract Test() {
+    function test() {
+        int d = date("02-16-2021 05:30:00 PM");
```

Comment:
This doesn't look like a valid ISO 8601 date string. Can we change the implementation to oonly accept ISO 8601 date strings, and then move this file to the ParseError tests alongside the one you already have.

Reply:
At first, I thought that it would be easier to leave it up to the parse function to digest strings so the user could enter dates however they wanted (`November 10, 2010`). Upon research, it looks like different interpreters will use either local time or UTC when parsing which can lead to confusion later.  Probably for the better to force ISO-8601. For clarity, I added a second parse error that returns the date in proper format. Second test created: `invalid_iso_date.cash`

Reply:
With this implementation it only allows the "full" ISO 8601 string (e.g. `2021-02-23T12:00:00.000Z`).

We might want to allow different versions (that are still compliant with ISO 8601), such as `2021-02-23T12:00:00`. But at the same time it's also good to be the most strict we can be to discourage mistakes. What do you think makes sense?

Probably if we do enforce a *single* format, I'd be in favour of ISO 8601 without the milliseconds (e.g. `2021-02-23T12:00:00`) since we round it to seconds any way.

Reply:
I've thought about it more and I think it makes most sense to enforce a specific format, but the "full" ISO8601 doesn't make sense, because there's no concept of timezones and there's no millisecond precision. So let's go with enforcing this specific format: `2021-02-23T12:00:00`. Probably the best way to go would be to use a regex for that.

Reply:
Sorry I had not seen your previous comment. Yeah I agree, stricter will probably be less confusing. Single format ```YYYY-MM-DDTHH:MM:SS``` makes sense as best option.

Reply:
Added it in the latest commit.

Reply:
Perfect, thanks!